contracts/FUSD.cdc

```diff
@@ -1,147 +1,191 @@
+                    ftVaultData: self.resolveContractView(resourceType: nil, viewType: Type<FungibleTokenMetadataViews.FTVaultData>()) as! FungibleTokenMetadataViews.FTVaultData?
+                )
+            case Type<FungibleTokenMetadataViews.FTDisplay>():
+                let media = MetadataViews.Media(
+                        file: MetadataViews.HTTPFile(
+                        url: ""
+                    ),
+                    mediaType: ""
+                )
+                let medias = MetadataViews.Medias([media])
+                return FungibleTokenMetadataViews.FTDisplay(
+                    name: "Flow USD",
+                    symbol: "FUSD",
+                    description: "Deprecated version of Flow USD. Developers are advised to not use this contract any more",
+                    externalURL: MetadataViews.ExternalURL(""),
+                    logos: medias,
+                    socials: {}
+                )
+            case Type<FungibleTokenMetadataViews.FTVaultData>():
+                return FungibleTokenMetadataViews.FTVaultData(
+                    storagePath: /storage/fusdVault,
+                    receiverPath: /public/fusdReceiver,
+                    metadataPath: /public/fusdBalance,
+                    receiverLinkedType: Type<&FUSD.Vault>(),
+                    metadataLinkedType: Type<&FUSD.Vault>(),
+                    createEmptyVaultFunction: (fun(): @{FungibleToken.Vault} {
+                        return <-FUSD.createEmptyVault(vaultType: Type<@FUSD.Vault>())
+                    })
+                )
+            case Type<FungibleTokenMetadataViews.TotalSupply>():
+                return FungibleTokenMetadataViews.TotalSupply(
+                    totalSupply: FUSD.totalSupply
+                )
+        }
+        return nil
+    }
 
     // Vault
     //
-    // Each user stores an instance of only the Vault in their storage
-    // The functions in the Vault are governed by the pre and post conditions
-    // in FungibleToken when they are called.
-    // The checks happen at runtime whenever a function is called.
-    //
-    // Resources can only be created in the context of the contract that they
-    // are defined in, so there is no way for a malicious user to create Vaults
-    // out of thin air. A special Minter resource needs to be defined to mint
-    // new tokens.
-    //
-    pub resource Vault: FungibleToken.Provider, FungibleToken.Receiver, FungibleToken.Balance {
+    access(all) resource Vault: FungibleToken.Vault {
 
         // holds the balance of a users tokens
-        pub var balance: UFix64
+        access(all) var balance: UFix64
 
         // initialize the balance at resource creation time
         init(balance: UFix64) {
             self.balance = balance
         }
 
+        /// Called when a fungible token is burned via the `Burner.burn()` method
+        access(contract) fun burnCallback() {
```

Comment:
Haven't looked into burn much, but it is possible for vaults with FUSD to be destroyed and this not be called, is that correct? If that is true, is this burn callback worth having since it is more than likely not accurate?

Reply:
yes. it is still better than nothing

Reply:
IMO nothing is better than the misleading one - and off-chain indexer sources can likely be more accurate than this. But I don't feel super strongly about that, so it's alright.